src/components/ui4/PromoCardsUi4.tsx

```diff
@@ -0,0 +1,109 @@
+
+interface Props {
+  navigation: NavigationBase
+  screenWidth: number
+}
+
+export const PromoCardsUi4 = (props: Props) => {
+  const { navigation, screenWidth } = props
+  const theme = useTheme()
+
+  const [promos, setPromos] = React.useState<PromoCard2[]>([])
+
+  // Check for PromoCard2 from info server:
+  React.useEffect(() => {
+    fetchInfo(`v1/promoCards2/${config.appId ?? 'edge'}`)
+      .then(async res => {
+        if (!res.ok) {
+          console.error(await res.text())
+          return
+        }
+        const infoData = await res.json()
+        const infoPromoCards = asArray(asPromoCard2)(infoData)
+
+        const buildNumber = getBuildNumber()
+        const countryCode = await getCountryCodeByIp()
+        const osType = Platform.OS.toLowerCase()
+        const osVersion = Platform.OS === 'ios' ? Platform.Version : `api${Platform.Version}`
+        const version = getVersion()
+        const currentDate = new Date()
+
+        setPromos(
+          infoPromoCards.filter(infoPromoCard => {
+            const { appVersion, minBuildNum, maxBuildNum, exactBuildNum, countryCodes, excludeCountryCodes, osTypes, osVersions, startIsoDate, endIsoDate } =
+              infoPromoCard
+
+            // Validate app version
+            // Ignore everything else if build version is specified and mismatched.
+            if (exactBuildNum != null && exactBuildNum !== buildNumber) return false
+
+            // Ignore min/max if app version is specified and mismatched.
+            if (appVersion != null && appVersion !== version) return false
+
+            // Look at min/max only if exact build or app version is not specified.
+            const isMinVersion = minBuildNum == null || minBuildNum <= buildNumber
+            const isMaxVersion = maxBuildNum == null || maxBuildNum >= buildNumber
+            if (!isMinVersion || !isMaxVersion) return false
+
+            // Validate country
+            const isCountryInclude =
+              countryCodes == null ||
+              countryCodes.length === 0 ||
+              countryCodes.map(countryCode => countryCode.toLowerCase()).includes(countryCode.toLowerCase())
+            const isCountryExclude =
+              excludeCountryCodes != null &&
+              excludeCountryCodes.length > 0 &&
+              excludeCountryCodes.map(countryCode => countryCode.toLowerCase()).includes(countryCode.toLowerCase())
+            if (!isCountryInclude || isCountryExclude) return false
+
+            // Validate OS type
+            if (osTypes != null && osTypes.length > 0 && !osTypes.map(osType => osType.toLowerCase()).includes(osType)) return false
+
+            // Validate OS version
+            if (osVersions != null && osVersions.length > 0 && !osVersions.includes(osVersion.toString())) return false
+
+            // Validate date range
+            const startDate = startIsoDate != null && !isNaN(Date.parse(startIsoDate)) ? new Date(startIsoDate) : null
```

Comment:
Iso dates can be lexographically compared. No need to convert with `new Date`. Just do

```
const currentIsoDate = new Date().toISOString()
if (startIsoDate > currentIsoDate) return false
if (endIsoDate < currentIsoDate) return false
```

Reply:
I don't want to assume that the dates given in the info data are correctly formed.